appserver/web/war-util/src/main/java/org/glassfish/web/loader/WebappClassLoader.java

```diff
@@ -2711,7 +2729,7 @@ protected ResourceEntry findResourceInternal(String name, String path) {
         entry = findResourceInternalFromRepositories(name, path);
 
         if (entry == null) {
-            synchronized (jarFiles) {
```

Comment:
Wouldn't be better to extract the synchronized block to a synchronized method, so it would use the same "default" lock?
Or all accesses to the final field should use own but same lock ... or use Concurrent implementation if it would make sense (I don't think it does).

Reply:
I don't want to change the synchronization fundamentally since there seems a lot of synchronization logic in the code. And changing those things means that we probably break the server functionality.